components/FeedList/FeedList.jsx

```diff
@@ -4,16 +4,26 @@ import {FlatList, View} from 'react-native';
 import Photos from '../../screens/ProfileScreen/page/Photos/Photos';
 import SkeletonPhotos from '../Skeleton/SkeletonPhotos';
 import SkeletonPosts from '../Skeleton/SkeletonPosts';
-import Warning from '../Warning/Warning';
 import FeedItem from './FeedItem';
 import styles from './style';
+import {memo, useState} from 'react';
+import Search from '../Search/Search';
+import Warning from '../Warning/Warning';
+import useDataFromAPI from '../../customHooks/UseDataFromAPI';
+import useDataForUpdate from '../../customHooks/useDataForUpdate';
 
-const FeedList = ({state, navigation, loading, showHide}) => {
+const FeedList = ({navigation, loading, showPostsOrPhotos}) => {
   const route = useRoute();
+  const [feedData, setFeedData] = useState([]);
+  const [filteredData, setFilteredData] = useState([]);
+
+  const {data, error} = useDataFromAPI('feeds');
+
+  useDataForUpdate(data, setFeedData, error);
 
   const renderSwitchValue = (item, index) => {
```

Comment:
useDataFromAPI and useDataForUpdate are already custom hooks that should manage their own dependencies and updates internally. Calling these hooks inside FeedList would create new function instances on every render, but it is perfectly acceptable because the hooks themselves handle data processing and caching internally.